### PR TITLE
Add log context to certificate refresh code

### PR DESCRIPTION
**Pull request by paulgb**

### CODE REVIEW

**Review by rolyatmax — APPROVED** [review]
Aside from the questions I left, this LGTM.

File: src/drone/proxy/certs.rs [hunk 1]
```diff
@@ -75,7 +75,9 @@ impl CertRefresher {
         };
 
         for path in key_cert_path_pair.parent_paths() {
-            watcher.watch(&path, RecursiveMode::NonRecursive)?;
+            watcher
+                .watch(&path, RecursiveMode::NonRecursive)
+                .context("Error installing watcher.")?;
```
Comment (rolyatmax) [inline src/drone/proxy/certs.rs:80]: Where exactly do these `context` strings go when the `?` "fails"?

File: src/drone/proxy/certs.rs [hunk 2]
```diff
@@ -1,5 +1,5 @@
 use crate::keys::{load_certs, load_private_key, KeyCertPathPair};
-use anyhow::Result;
+use anyhow::{Context, Result};
```
[the remark this answers is not in the document]
Comment (paulgb) [reply src/drone/proxy/certs.rs:2]: Yes, in Rust you need to import a trait to use its methods (in this case `context`).

File: src/drone/proxy/certs.rs (hunk 1 above)
Comment (paulgb) [reply src/drone/proxy/certs.rs:80]: When an error is reported (technically, when its debug formatter is called), the context message is added to the output along with the original error.

**Pull request merged**